shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/viewModel/FavoritesViewModel.kt

```diff
@@ -0,0 +1,186 @@
+    sealed interface Event {
+        data object ReloadFavorites : Event
+
+        data class SetActive(val active: Boolean, val isReturningFromBackground: Boolean?) : Event
+
+        data class SetAlerts(val alerts: AlertsStreamDataResponse?) : Event
+
+        data class SetLocation(val location: Position?) : Event
+
+        data class SetNow(val now: Instant) : Event
+    }
+
+    data class State(
+        val favorites: Set<RouteStopDirection>?,
+        val isReturningFromBackground: Boolean,
+        val routeCardData: List<RouteCardData>?,
+        val staticRouteCardData: List<RouteCardData>?,
+    ) {
+        constructor() : this(null, false, null, null)
+    }
+
+    @Composable
+    override fun runLogic(events: Flow<Event>): State {
+        var favorites: Set<RouteStopDirection>? by remember { mutableStateOf(null) }
+        var isReturningFromBackground: Boolean by remember { mutableStateOf(false) }
+        var routeCardData: List<RouteCardData>? by remember { mutableStateOf(null) }
+        var staticRouteCardData: List<RouteCardData>? by remember { mutableStateOf(null) }
+
+        var active: Boolean by remember { mutableStateOf(true) }
+        var alerts: AlertsStreamDataResponse? by remember { mutableStateOf(null) }
+        var location: Position? by remember { mutableStateOf(null) }
+        var now: Instant by remember { mutableStateOf(Clock.System.now()) }
+
+        val globalData = getGlobalData("FavoritesViewModel.getGlobalData")
+        val stopIds =
+            remember(favorites, globalData) {
+                val stops = favorites?.mapNotNull { globalData?.getStop(it.stop) }
+                stops?.flatMap { stop ->
+                    stop.childStopIds.filter { globalData?.stops?.containsKey(it) ?: false } +
+                        stop.id
+                }
+            }
+        val schedules = getSchedules(stopIds, "FavoritesViewModel.getSchedules")
+        val predictions = subscribeToPredictions(stopIds, active)
+
+        LaunchedEffect(Unit) { favorites = favoritesUsecases.getRouteStopDirectionFavorites() }
+
+        LaunchedEffect(Unit) {
+            events.collect { event ->
+                when (event) {
+                    Event.ReloadFavorites ->
+                        favorites = favoritesUsecases.getRouteStopDirectionFavorites()
+                    is Event.SetActive -> {
+                        active = event.active
+                        event.isReturningFromBackground?.let { isReturningFromBackground = it }
+                    }
+                    is Event.SetAlerts -> alerts = event.alerts
+                    is Event.SetLocation -> location = event.location
+                    is Event.SetNow -> now = event.now
+                }
+            }
+        }
+
+        LaunchedEffect(predictions) {
+            if (predictions != null) {
+                isReturningFromBackground = false
+            }
+        }
```

Comment:
question: why is this false? Maybe something about the naming of `active` and `isReturningFromBackground` could be clarified through modified naming or comments.

Reply:
This is false because once the predictions have loaded, the favorites tab has finished returning from the background. On iOS, the `NearbyTransitView` sets `isReturningFromBackground = false` (which is bound to the `ErrorBannerViewModel`’s `loadingWhenPredictionsStale`) whenever a new predictions message is received, and on Android, the `PredictionsViewModel` directly sets the `ErrorBannerViewModel`’s `loadingWhenPredictionsStale` when a new `PredictionsByStopJoinResponse` is received.

Renamed `isReturningFromBackground` to `awaitingPredictionsAfterBackground` and tweaked the `SetActive` event.

Reply:
That renaming is great, thank you!